src/Summoner/Tui.hs

```diff
@@ -200,53 +106,19 @@ app :: [FilePath] -> App (Form SummonKit e SummonForm) e SummonForm
 app dirs = App
     { appDraw = draw
     , appHandleEvent = \s ev -> case ev of
-        VtyEvent V.EvResize {}       -> continue s
+        VtyEvent V.EvResize {} -> continue s
         -- TODO: should we cancel on Esc and apply on Ctrl+Enter?
         VtyEvent (V.EvKey V.KEnter [V.MCtrl]) -> halt s
         VtyEvent (V.EvKey V.KEsc []) -> halt s
-        VtyEvent (V.EvKey (V.KChar 'd') [V.MCtrl]) -> do
-            s' <- handleFormEvent ev s
-            continue $
-                if focusGetCurrent (formFocus s') == Just UserOwner
-                then mkForm $ formState s' & user . owner .~ ""
-                else s'
-
+        VtyEvent (V.EvKey (V.KChar 'd') [V.MCtrl]) ->
+            handleFormEvent ev s >>= continue . summonFormValidation dirs . ctrlD
         MouseDown n _ _ _ -> case n of
             GitHubEnable  -> handleFormEvent ev s >>= continue . mkForm . formState
             GitHubDisable -> handleFormEvent ev s >>= continue . mkForm . formState
             _             -> handleFormEvent ev s >>= continue
```

Comment:
The pattern with `handleFormEvent ev s >>= continue . f` appears in many places. Now, after refactoring fields validation into separate function it's more visually clear. Maybe it makes sense to implement some function like:

```haskell
withForm ev s f = handleFormEvent ev s >>= continue . f
```

What do you think? Doesn't really decrease the number of lines, but maybe makes existing lines shorter...